### ISSUE TITLE
[Bug]: Deleted accounts remain as a follower when they shouldn't be

### ISSUE THREAD

**Issue by arunsathiya** [Automattic/wp-calypso#56919]
### Quick summary

When a WordPress.com user account is deleted, the account remains as a follower on the sites that they followed. **I think** this is unexpected. When an account is deleted, it's assumed that all subscriptions are removed along with it.

If this is working as expected today, the report can be closed. 

### Steps to reproduce

- Start with a simple and an Atomic (WordPress.com sites with a plugin or third-party theme installed) site.
- Invite another WordPress.com user account as a follower. Details of the roles here: https://wordpress.com/support/user-roles/#adding-users-to-your-site
- Once they accept the invitation, have them delete account at https://wordpress.com/me/account
- See that they remain as a follower both on simple and Atomic, when they shouldn't be.

| Atomic | Simple |
|------|------|
| <img width="768" alt="Screenshot 2021-10-09 at 1 11 05 PM" src="https://user-images.githubusercontent.com/18581859/136649335-69a5589c-5115-48c1-9366-32202f375816.png"> | <img width="775" alt="Screenshot 2021-10-09 at 1 11 08 PM" src="https://user-images.githubusercontent.com/18581859/136649332-b1036959-08b9-48bf-aaaa-f9184a0fc550.png"> | 

In these images, `physiopsychicnonprimitive` has deleted the account. But the account remains as a follower.

New post emails are sent too. I don't think this is expected.

| Atomic | Simple |
|------|------|
| <img width="725" alt="Screenshot 2021-10-09 at 1 17 33 PM" src="https://user-images.githubusercontent.com/18581859/136649403-5a477ab4-c792-4272-827e-0fe15aa96bf8.png"> | <img width="731" alt="Screenshot 2021-10-09 at 1 17 39 PM" src="https://user-images.githubusercontent.com/18581859/136649400-818de1f2-ec45-4c3e-8158-04b88cae70c7.png"> | 

### What you expected to happen

Deleted account must be removed from the sites that they followed and must stop receiving emails.

### What actually happened

Deleted accounts remain as a follower and continue to receive emails.

### Context

Manual testing.

### Browser

Google Chrome/Chromium, Mozilla Firefox, Microsoft Edge, Apple Safari, iOS Safari, Android Chrome

### Simple/Atomic

Simple, Atomic

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

Yes, easy to implement

### Workaround details

Admin can remove the followers from the sites: https://wordpress.com/people/team 

<img width="788" alt="image" src="https://user-images.githubusercontent.com/18581859/136649478-40550873-7dde-4d7d-9848-aa4438d9236b.png">

**Comment by mreishus** [comment]
Working on a fix in D69110-code

**Comment by mreishus** [comment]
After 30 days, a deleted user's account is purged. Open question: Does this delete the row belonging to the user in the subscription matrix? After our initial look at `wpcom_purge_a_group_of_users()` we're thinking it doesn't, but this warrants more investigation.

**Comment by mreishus** [comment]
When users are purged after 30 days, the rows in the subscriptions matrix belonging to them aren't deleted.

**Comment by mreishus** [comment]
D69110-code deployed.
* :heavy_check_mark: Deleted accounts will no longer appear while looking through the followers list.
* :heavy_check_mark: Sites with a smaller number of followers will no longer include deleted accounts in in follower counts.
* :x: Sites with a larger number of followers will continue to include deleted accounts in follower counts. However, they still won't be visible wile looking through the followers list. This represents a more involved engineering problem. There is some discussion and steps to move forward in this internal document: pdgA0m-71-p2
* :x: New post emails will continue to be sent. This can either be fixed when the large counts are fixed above, or we can add an additional filter on the outgoing subscriptions email that filters out deleted users. 

For now, I need to put this issue aside, but this is where we're at and how to move forward.

**Comment by allilevine** [comment]
> How big is the impact of new post emails being sent to these deleted users?

@candy02058912 This seems like the higher priority remaining issue to me. Can the users unsubscribe from the email?

**Comment by candy02058912** [comment]
> Can the users unsubscribe from the email?

I got a bit confused by the question thinking they've already been deleted, but I'm thinking you mean will they keep getting the emails although they've been deleted from WordPress.com?

Honestly, I don't know, and it might be hard to test because, as far as I'm reading this, it only happens when there's a large follower count. @mreishus am I reading it correctly? 🙏

**Comment by allilevine** [comment]
> I got a bit confused by the question thinking they've already been deleted, but I'm thinking you mean will they keep getting the emails although they've been deleted from WordPress.com?

Yes. Is there a way for them to stop getting the emails, since the emails don't stop when they're deleted?

**Comment by mreishus** [comment]
The incorrect emails will be sent even if there is a small follower count. For small follower count sites, the change is that the "total" number displayed will be adjusted downward here: D69110-code. To stop the emails, we'd need to implement the purging changes talked about in pdgA0m-71-p2 or add a filter on the outgoing subscription emails that checks and blocks.

**Comment by mmtr** [comment]
@BogdanUngureanu was working on this a couple of years ago and opened D82698-code to hopefully fix it. However, it seems that more extra work is needed in that diff.

**Comment by lsl** [comment]
> add a filter on the outgoing subscription emails that checks and blocks.

This sounds like the right approach as a first step.

By checks, you mean, check the user exists and isn't deleted?

**Comment by lsl** [comment]
I've created https://github.com/Automattic/wp-calypso/issues/93548 to track the email block fix, I'm moving this issue to needs shaping with a lower priority as the count being wrong seems pretty minor but something we should probably fix if possible.

**Comment by kraftbj** [comment]
It's also odd to see purged users from a site owner's POV:
![2024-09-16 at 20 24](https://github.com/user-attachments/assets/6df4dfab-9d78-4a70-8857-5bbf26fb4709)

I can appreciate not removing them when they're initially deleted, but when deleting an account, it says content will be deleted after 30 days. Are the sites I am subscribed to part of my "content"?

In either event, as a site owner, if I see a handful of these, I begin not to trust the system if these "unsubscribes" aren't being understood as such.
